packages/dev/core/src/Materials/effect.ts

```diff
@@ -1448,6 +1448,17 @@ export class Effect implements IDisposable {
         return this;
     }
 
+    /**
+     * Use this wisely: It will remove the cached code from this effect
+     * It is probably ok to call it if you are not using ShadowDepthWrapper or if everything is already up and running

```

Comment:
I think we should also add a comment stating that if lost contexts are to be managed, this method should not be called as we will need the shader source code to recreate the effects on context restoration.